src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs

```diff
@@ -401,18 +401,13 @@ public XmlDocument UpdatePackageSources(XmlDocument nugetConfig, Dictionary<stri
                 }
                 else if (currentNode.NodeType == XmlNodeType.Comment)
                 {
-                    if (currentNode.Value.Equals(MaestroBeginComment, StringComparison.OrdinalIgnoreCase) ||
-                        currentNode.Value.Equals(MaestroEndComment, StringComparison.OrdinalIgnoreCase))
+                    if (currentNode.Value.Equals(MaestroBeginComment, StringComparison.OrdinalIgnoreCase))
                     {
-                        withinMaestroComments = currentNode.Value.Equals(MaestroBeginComment, StringComparison.OrdinalIgnoreCase);
-                        currentNode = RemoveCurrentNode(currentNode);
-                        continue;
+                        withinMaestroComments = true;
```

Comment:
What happens here if someone malforms the comment? e.g. they manually delete the end comment but not the beginning, or vice-versa.

Reply:
- if it's not valid XML, we never get this far
- If they delete either comment, the code the matching checks like this catch them:

```
            XmlComment startCommentBlock = GetFirstMatchingComment(disabledSourcesNode, MaestroBeginComment);
            if (startCommentBlock != null)
            {	            {
                var managedSources = GetManagedPackageSources(maestroManagedFeedsByRepo[repoName]).OrderByDescending(t => t.feed).ToList();	                insertAfterNode = startCommentBlock;
            }
```

I can easily add a test case or two for this scenario too.